Add docs regarding dependency defs (#315)
AlaskaAirlines/WC-Generator: docs/bestPractices.md

The best strategy is the overly communicated strategy. The more you can project the intent of upcoming releases, the more users will not be caught off guard and have an opportunity to control the eventual rollout of the updated thing.

## Direct or peer dependency elements 

> Much of the confusion lies within how `CustomElementRegistry.define()` works. When defining a custom element to the browser, load order matters. Using the `customElements.get` method, at the point of registration to see if the element has already been registered. If not, register it.

Reviewer: Maybe link [MDN](https://developer.mozilla.org/en-US/docs/Web/API/CustomElementRegistry/define) here?

```js
// define the name of the custom component
if (!customElements.get("auro-[name]")) {
  customElements.define("auro-[name]", Auro[Name]);
}
```

> At this time, the `customElements.get()` method only returns constructor of the element and there currently is no way to natively determine what version of the element had been registered or what versions have requested to be registered. 

Reviewer: ```suggestion
Currently, the `customElements.get()` method only returns the constructor of the element, and there is no way to natively determine what version of the element was registered. 
```
